### PR TITLE
SDIT-1328 court case create

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 1]
```diff
@@ -438,3 +513,20 @@ data class SentenceTermResponse(
...
+@Schema(description = "Court case create request")
+data class CreateCourtCaseRequest(
+  @Schema(description = "Court case start date", required = true)
+  val startDate: LocalDate, // the warrant date in sentencing
+  @Schema(description = "Legal Case type", required = true)
+  val legalCaseType: String, // either A for Adult or sentences to create new type
+  @Schema(description = "Court Id (establishment)", required = true)
+  val court: String, // either A for Adult or sentences to create new type
+  @Schema(description = "Case status", required = true)
+  val status: String, // ACTIVE, INACTIVE, CLOSED
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:526]: will remove some of these code notes later, they are useful during analysis

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 2]
```diff
@@ -58,6 +67,31 @@ class SentencingService(
...
+  @Audit
+  fun createCourtCase(offenderNo: String, request: CreateCourtCaseRequest) =
+    findPrisoner(offenderNo).findLatestBooking().let { booking ->
+      val courtCaseId = courtCaseRepository.save(
+        CourtCase(
+          offenderBooking = booking,
+          legalCaseType = lookupLegalCaseType(request.legalCaseType),
+          beginDate = request.startDate,
+          caseStatus = lookupCaseStatus(request.status),
+          court = lookupEstablishment(request.court),
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:79]: will add in caseSequence in next commit - currently defaulting to 1

**Review by andymarke — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt (hunk 1 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:526]: yes - that's right. I'll probably remove the Schema for these internal syscon ones if its not adding much

**Pull request merged**